Author: Rasmus Thomsen <cogitri@exherbo.org>
Reason: Mesa's meson build system doesn't pass
NDEBUG to the cppargs when building with LLVM
without RTTI support enabled. This causes the
build to fail because some parts of debugging
need RTTI support.
Upstream: Created bug for it, https://bugs.freedesktop.org/show_bug.cgi?id=106391
diff --git a/meson.build b/meson.build
index 6e61766809..269b5bd3b9 100644
--- a/meson.build
+++ b/meson.build
@@ -1150,6 +1150,9 @@ if with_llvm
   # ensure that linking works.
   if dep_llvm.get_configtool_variable('has-rtti') == 'NO'
     cpp_args += '-fno-rtti'
+    pre_args += [
+      '-DNDEBUG',
+    ]
   endif
 elif with_amd_vk or with_gallium_radeonsi or with_gallium_swr
   error('The following drivers require LLVM: Radv, RadeonSI, SWR. One of these is enabled, but LLVM is disabled.')
